File: project/5.03_コンストラクタの追加/java/java_ex5_03_1/src/java_ex5_03_1/Ex5_03_1.java
package java_ex5_03_1;

class Person {
	String name;
	int age;
	float income;
	
	Person( String arg_name, int arg_age, float arg_income){
		name = arg_name;
		age = arg_age;
		income = arg_income;
	}
}

public class Ex5_03_1 {

	public static void main(String[] args) {

//		Person p1 = new Person();		// 引数なしコンストラクタを明示的に作成しないとエラー
//		p1.name = "Keroichi";
//		p1.age = 40;
//		p1.income = 20_000_000;
		
		Person p1 = new Person( "Keroichi", 40, 20_000_000 );
		
		System.out.println( p1.name + "は" );
		System.out.println( p1.age + "歳で" );
		System.out.println( p1.income + "円の収入です");
	}

}

File: project/5.03_コンストラクタの追加/py/py_ex5_03_1/src/ex5_03_1.py
'''
Created on 2023/04/30

@author: Keroichi.T
'''

class Person:
    name = ""
    age = 0
    income = 0.0

    def __init__(self, arg_name, arg_age, arg_income ):
        # ↓コンストラクタ内でのみ有効な変数となる
        # name = arg_name
        # age = arg_age
        # income = arg_income

        # ↓メソッド内からクラス変数を参照する場合
        Person.name = arg_name
        Person.age = arg_age
        Person.income = arg_income

def main():
    # 引数なしコンストラクタがないとエラー
    # p1 = Person()

    p1 = Person( "Keroichi", 40, 20_000_000 )    
    print( "{}は\n{}歳で\n{}円の収入です".format(p1.name, p1.age, p1.income ) )
    
    # コンストラクタ内でクラス変数指定して書き換えたので内容も更新されている
    print( "{}は\n{}歳で\n{}円の収入です".format(Person.name, Person.age, Person.income ) )

if __name__ == '__main__':
    main()